pkg/kubelet/nodeshutdown/systemd/inhibit_linux.go

```diff
@@ -47,6 +47,25 @@ type DBusCon struct {
 	SystemBus dBusConnector
 }
 
+func NewDBusCon() (*DBusCon, error) {
+	conn, err := dbus.SessionBusPrivate()
```

Comment:
`dbus.SystemBus()` is singleton, it can no longer be used after being closed.

I sent a patch to dbus. https://github.com/godbus/dbus/pull/232

Reply:
nice, I see the patch was accepted. Does it mean if we upgrade to new dbus version with that patch we should have "autoreconnect" logic then? :)

Reply:
This PR is still needed. That PR I sent to GoDbus can only fixed singleton reuse, but cannot implement the "autoreconnect" logic and still sense disconnection.

Reply:
got it, thanks for clarification. 

I'm still unclear on usage of session bus as opposed to system bus, i.e. using `dbus.SessionBusPrivate()` instead of `dbus.SystemBusPrivate()`. Shouldn't we be using the system bus specifically since logind is exposed on system bus?

Reply:
Makes sense, Let me ask if godbus/dbus has plans to release a new version. https://github.com/godbus/dbus/pull/232#issuecomment-805557363